geoips/pydantic/products.py

```diff
@@ -0,0 +1,231 @@
+# # # This source code is protected under the license referenced at
+# # # https://github.com/NRLMMD-GEOIPS.
+
+"""Order-based procflow Product building Models."""
+
+# Python Standard Libraries
+from typing import Any, Dict, List
+
+# Third-Party Libraries
+from pydantic import BaseModel, Field, model_validator, ConfigDict
+
+# GeoIPS imports
+from geoips import interfaces
+from geoips.pydantic.bases import PluginModel
+
+
+def get_plugin_names(plugin_type: str) -> List[str]:
+    """Retrieve a list of valid plugin names for a given plugin type.
+
+    Args
+    ----
+    plugin_type
+        The type of plugin
+
+    Returns
+    -------
+        list: A list of plugin names for a valid plugin type
+
+    Raises
+    ------
+        AttributeError: If the plugin type is invalid
```

Comment:
Not numpy style list

https://numpydoc.readthedocs.io/en/latest/format.html#raises